graph/GuidelinesGraph.md

```diff
@@ -332,6 +333,15 @@ For a complete mapping of error codes to HTTP statuses, see
 
 <a name="api-contract-and-non-backward-compatible-changes"></a>
 
+### Limitations on core types
```

Comment:
Can we add this guidance for applications and servicePrincipals too?  In my mind those are core types.

We actually now have a very real example of why this is important for downstream tooling.  Adding a structural property that is handled by a different workload (from the core workload) is really difficult to model with Bicep.  If instead it's added as a new entity + nav property, this fits cleanly with Bicep (and ARM's) nested child resources.

Reply:
> Adding a structural property that is handled by a different workload (from the core workload)

FYI, I think we have a *different* ADR that says we should stop using composite entities.

I'm ok adding `applications` and `servicePrincipals`, but I don't think those have been brought up with the API council. Do we want to take this PR as-is and do a follow-up to add apps and sps after discussion with the council?

Reply:
I am inclined to take Garrett's suggestion. If you are fine with this, I will merge as-is @dkershaw10